File: backend/models/equipos.go
package models

type Equipos struct {
	EquipoId             int    `json:"equipo_id"`
	Codigo               string `json:"codigo"`
	TipoEquipo           string `json:"tipo_equipo"`
	Modelo               string `json:"modelo"`
	NumeroSerie          string `json:"numero_serie"`
	Accesorios           string `json:"accesorios"`
	Descripcion          string `json:"descripcion_problema"`
	Observacion          string `json:"observacion"`
	FechaRecepcion       string `json:"fecha_recepcion"`
	FechaEstimadaEntrega string `json:"fecha_estimada_entrega"`
	FechaCreacion        string `json:"fecha_creacion"`
	FechaModificacion    string `json:"fecha_modificacion"`

	MarcaId   int `json:"marca_id"`
	ClienteId int `json:"cliente_id"`
	EstadoId  int `json:"estado_id"`
}

type EquiposDTO struct {
	EquipoId             int    `json:"equipo_id"`
	Codigo               string `json:"codigo"`
	TipoEquipo           string `json:"tipo_equipo"`
	Modelo               string `json:"modelo"`
	NumeroSerie          string `json:"numero_serie"`
	Accesorios           string `json:"accesorios"`
	Descripcion          string `json:"descripcion_problema"`
	Observacion          string `json:"observacion"`
	FechaRecepcion       string `json:"fecha_recepcion"`
	FechaEstimadaEntrega string `json:"fecha_estimada_entrega"`
	FechaCreacion        string `json:"fecha_creacion"`
	FechaModificacion    string `json:"fecha_modificacion"`

	MarcaId int    `json:"marca_id"`
	Marca   string `json:"marca"`

	EstadoId int    `json:"estado_id"`
	Estado   string `json:"estado"`

	ClienteId int    `json:"cliente_id"`
	Nombres   string `json:"nombres"`
	Apellidos string `json:"apellidos"`
}

File: frontend/src/modelos/equipos.ts
export interface Equipo {
    equipo_id: number;
    codigo: string;
    tipo_equipo: string;
    modelo: string;
    numero_serie: string;
    accesorios: string;
    descripcion_problema: string;
    observacion: string;
    fecha_recepcion: string;
    fecha_estimada_entrega: string;
    fecha_creacion: string;
    fecha_modificacion: string;
    marca_id: number;
    cliente_id: number;
    estado_id: number;
    usuario_id: number;
}

export interface EquipoDTO {
    equipo_id: number;
    codigo: string;
    tipo_equipo: string;
    modelo: string;
    numero_serie: string;
    accesorios: string;
    descripcion_problema: string;
    observacion: string;
    fecha_recepcion: string;
    fecha_estimada_entrega: string;
    fecha_creacion: string;
    fecha_modificacion: string;
    marca_id: number;
    marca: string;
    estado_id: number;
    estado: string;
    cliente_id: number;
    nombres: string;
    apellidos: string;
}